exporter/kafkaexporter/config.go

```diff
@@ -74,5 +79,29 @@ var _ config.Exporter = (*Config)(nil)
 
 // Validate checks if the exporter configuration is valid
 func (cfg *Config) Validate() error {
+	switch strings.ToLower(cfg.Compression.Codec) {
+	case "none":
+		if cfg.Compression.Level != defaultCompressionLevel {
+			return fmt.Errorf("invalid compression level %d: should be %d when not using compression", cfg.Compression.Level, defaultCompressionLevel)
+		}
+	case "gzip":
+		if cfg.Compression.Level > 9 || cfg.Compression.Level < 1 {
```

Comment:
Can you point me to where these max levels defined?

Reply:
Yes. The compression level is up to the compression libraries. [gzip](https://www.gnu.org/software/gzip/manual/gzip.html#Invoking-gzip), [lz4](https://cwiki.apache.org/confluence/display/KAFKA/KIP-390%3A+Allow+fine-grained+configuration+for+compression#KIP390:Allowfinegrainedconfigurationforcompression-Compressionlevel), [zstd](https://facebook.github.io/zstd/zstd_manual.html#Chapter1)

Reply:
Looking at the document [here](https://cwiki.apache.org/confluence/display/KAFKA/KIP-390%3A+Support+Compression+Level) appears to suggest some defaults, would it be worth using those if the user choose to leave the level to default?

Reply:
If the user leave the level to default, we will pass `-1000` to Sarama. And Sarama will use the default compression level corresponding to the compression method (e.g. `6` fot `gzip`).

Reply:
Right, this makes sense, but validation here checks the configuration level, not the sarama value correct?

Reply:
Validation method here needs to check if the compression level is as default, if it's the default value, no error should be caused. I have fixed this. Please see the latest commit.